$OpenBSD: patch-gnome-session_gsm-xsmp-client_c,v 1.2 2010/04/22 22:38:51 ajacoutot Exp $

GNOME Bugzilla #598211
Fix SIGSEGV in SmsDie(): throw an error in xsmp_stop() when
client->priv->conn is NULL.

--- gnome-session/gsm-xsmp-client.c.orig	Fri Apr 23 00:33:12 2010
+++ gnome-session/gsm-xsmp-client.c	Fri Apr 23 00:33:12 2010
@@ -725,6 +725,15 @@ xsmp_stop (GsmClient *client,
                 return FALSE;
         }
 
+        /*FIXME: What do we do if the client can't be stoped ?
+         * (probably a cleaner solution exists...) */
+        if (xsmp->priv->conn == NULL) {
+                g_set_error (error,
+                             GSM_CLIENT_ERROR,
+                             GSM_CLIENT_ERROR_NOT_REGISTERED,
+                             "Client is not registered");
+                return FALSE;
+        }
         SmsDie (xsmp->priv->conn);
 
         return TRUE;
